src/main/kotlin/org/eln2/mc/common/blocks/foundation/MultipartBlock.kt

```diff
@@ -0,0 +1,1356 @@
+
+    private val epsilon = 0.00001
+    private val emptyBox = box(0.0, 0.0, 0.0, epsilon, epsilon, epsilon)
+
+    //#region Block Methods
+
+    override fun newBlockEntity(pPos: BlockPos, pState: BlockState): BlockEntity {
+        return MultipartBlockEntity(pPos, pState)
+    }
+
+    @Deprecated("Deprecated in Java", ReplaceWith("true"))
+    override fun skipRendering(pState: BlockState, pAdjacentBlockState: BlockState, pDirection: Direction): Boolean {
+        return true
+    }
+
+    @Deprecated("Deprecated in Java")
+    override fun getCollisionShape(
+        pState: BlockState,
+        pLevel: BlockGetter,
+        pPos: BlockPos,
+        pContext: CollisionContext,
+    ): VoxelShape {
+        return getMultipartShape(pLevel, pPos, pContext)
+    }
+
+    @Deprecated("Deprecated in Java")
+    override fun getShape(
+        pState: BlockState,
+        pLevel: BlockGetter,
+        pPos: BlockPos,
+        pContext: CollisionContext,
+    ): VoxelShape {
+        return getPartShape(pLevel, pPos, pContext)
+    }
+
+    @Deprecated("Deprecated in Java")
+    override fun getVisualShape(
+        pState: BlockState,
+        pLevel: BlockGetter,
+        pPos: BlockPos,
+        pContext: CollisionContext,
+    ): VoxelShape {
+        return getPartShape(pLevel, pPos, pContext)
+    }
+
+    override fun onDestroyedByPlayer(
+        state: BlockState?,
+        level: Level?,
+        pos: BlockPos?,
+        player: Player?,
+        willHarvest: Boolean,
+        fluid: FluidState?,
+    ): Boolean {
+        if (pos == null) {
+            return false
+        }
+
+        if (level == null) {
+            return false
+        }
+
+        if (player == null) {
+            return false
+        }
+
+        val multipart = level.getBlockEntity(pos) as? MultipartBlockEntity
+
+        if (multipart == null) {
+            LOG.error("Multipart null at $pos")
+            return false
+        }
+
+        val saveTag = CompoundTag()
+
+        val removedId = multipart.remove(player, level, saveTag)
+            ?: return false
+
+        if (!player.isCreative) {
+            player.inventory.add(Part.createPartDropStack(removedId, saveTag))
+        }
+
+        // We want to destroy the multipart only if it is empty
+        val multipartIsDestroyed = multipart.isEmpty
+
+        if (multipartIsDestroyed) {
+            level.destroyBlock(pos, false)
+        }
+
+        return multipartIsDestroyed
+    }
+
+    override fun addRunningEffects(state: BlockState?, level: Level?, pos: BlockPos?, entity: Entity?): Boolean {
+        return true
+    }
+
+    override fun addLandingEffects(
+        state1: BlockState?,
+        level: ServerLevel?,
+        pos: BlockPos?,
+        state2: BlockState?,
+        entity: LivingEntity?,
+        numberOfParticles: Int,
+    ): Boolean {
+        return true
+    }
+
+    @Deprecated("Deprecated in Java")
```

Comment:
I don't know that we need to override the annotations when overriding the functions unless we want to be telling ourselves to deprecating code that we call ourselves. I'd remove this since we're developing this codebase more than maintaining it, unless we absolutely need it.